src/cr/cube/dimension.py

```diff
@@ -251,222 +251,30 @@ class Dimension(object):
-
-        This ordering corresponds to how value "rows" (or columns) are to
-        appear after subtotals have been inserted at their anchor locations.
-        Where more than one subtotal is anchored to the same location, they
-        appear in their document order in the cube response.
-
-        Only elements in the passed *elements* collection appear, which
-        allows control over whether missing elements are included by choosing
-        `.all_elements` or `.valid_elements`.
-        """
-        subtotals = self._subtotals
-
-        for subtotal in subtotals.iter_for_anchor("top"):
-            yield subtotal
-
-        for element in elements:
-            yield element
-            for subtotal in subtotals.iter_for_anchor(element.element_id):
-                yield subtotal
-
-        for subtotal in subtotals.iter_for_anchor("bottom"):
-            yield subtotal
-
-    @lazyproperty
-    def _subtotals(self):
-        """_Subtotals sequence object for this dimension.
-
-        The subtotals sequence provides access to any subtotal insertions
-        defined on this dimension.
+        Elements in this sequence appear in cube-result order. Display order (including
+        resolution of the explicit-reordering transforms cascade) is provided by
+        a separate `.display_order` attribute on _AllElements.
         """
-        view = self._dimension_dict.get("references", {}).get("view", {})
-        # ---view can be both None and {}, thus the edge case.---
-        insertion_dicts = (
-            [] if view is None else view.get("transform", {}).get("insertions", [])
-        )
-        return _Subtotals(insertion_dicts, self.valid_elements)
-
-
-class NewDimension(Dimension):
-    """A new version of Dimension that knows about its analysis-level transforms.
-
-    The original Dimension object knew about its *base*-transforms. This new version
-    adds the cascading behavior of analysis-specific transforms that elaborate those
-    appearing on the cube-response itself.
-
-    The idea is that soon we can just merge the properties of this class into the main
-    Dimension class and get rid of this one. That will have to wait until FrozenSlice
-    loads its own data and dimensions from the cube response rather than relying on
-    CrunchCube to do that.
-    """
-
-    def __init__(self, dimension, dimension_transforms_dict):
-        super(NewDimension, self).__init__(
-            dimension._dimension_dict, dimension._dimension_type
+        return _AllElements(
+            self._dimension_dict["type"], self._dimension_transforms_dict
         )
-        self._dimension_transforms_dict = dimension_transforms_dict
 
-    @lazyproperty
-    def all_elements(self):
-        """_NewAllElements object providing cats or subvars of this dimension.
+    def apply_transforms(self, dimension_transforms):
+        """Return a new `Dimension` object with `dimension_transforms` applied.
```

Comment:
So there was an interesting problem in this process, which was where to apply transforms. Transforms are known at Cube construction time, passed in as a parameter. The question is where to apply them to the dimensions for individual slices.

My first instinct was to let `AllDimensions` do the job; transforms are on dimensions and AllDimensions creates all the dimensions, so it makes sense it should parcel out the transforms on first construction. The problem that arose is that `AllDimensions` then needed to duplicate all the rules that determine which dimension later becomes the rows-dimension in a slice and which becomes the columns-dimension, if any.

That was more than I could stomach after I thought about it awhile, just too brittle. So the solution I came up with is that all dimensions are initially constructed without transforms (defaults to None/{}). This takes `AllDimensions` out of the transforms assignment role. Then, `FrozenSlice` gets the right dimensions it wants and calls this method on each to produce the same dimension, but with transforms applied. The returned dimension object is different, so still immutable, but the same in every way except this time it has the desired transforms.

Anyway, thought that was a little bit interesting so thought I would mention.